pkg/http/prometheus/prometheus.go

```diff
@@ -0,0 +1,56 @@
+package prometheus
+
+import (
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/felixge/httpsnoop"
+	"github.com/gorilla/mux"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+type middleware struct {
+	buckets           []float64
+	httpTotalRequests *prometheus.CounterVec
+	httpDuration      *prometheus.HistogramVec
+}
+
+func New(opts ...Option) func(http.Handler) http.Handler {
+	m := &middleware{
+		buckets: []float64{0.5, 1, 2, 3, 5},
+	}
+
+	for _, opt := range opts {
+		opt(m)
+	}
+
+	m.httpDuration = promauto.NewHistogramVec(prometheus.HistogramOpts{
+		Name:    "http_duration_seconds_bucket",
```

Comment:
nit: not sure if this really matters but by [convention](https://prometheus.io/docs/practices/naming/) the suffix should just be the unit `http_duration_seconds`. When looking at /metrics I see that prometheus appended type to suffix already:

```
# HELP http_duration_seconds_bucket Duration of HTTP requests.
# TYPE http_duration_seconds_bucket histogram
http_duration_seconds_bucket_bucket{code="200",method="get",path="/",le="0.5"} 1
http_duration_seconds_bucket_bucket{code="200",method="get",path="/",le="1"} 1
http_duration_seconds_bucket_bucket{code="200",method="get",path="/",le="2"} 1
http_duration_seconds_bucket_bucket{code="200",method="get",path="/",le="3"} 1
http_duration_seconds_bucket_bucket{code="200",method="get",path="/",le="5"} 1
http_duration_seconds_bucket_bucket{code="200",method="get",path="/",le="+Inf"} 1
http_duration_seconds_bucket_sum{code="200",method="get",path="/"} 0.000258
http_duration_seconds_bucket_count{code="200",method="get",path="/"} 1
```

Reply:
Good catch. I did not notice that there was a duplicate `_bucket`.